argparse_test.go

```diff
@@ -792,6 +1035,6 @@ func TestSelectorDefaultValueFail(t *testing.T) {
 
 	// Should pass on failure
 	if err == nil || err.Error() != "cannot use default type [bool] as type [string]" {
-		t.Errorf("Test %s failed: expected error [%s], got error [%s]", t.Name(), "cannot use default type [bool] as type [string]", err.Error())
+		t.Errorf("Test %s failed: expected error [%s], got error [%+v]", t.Name(), "cannot use default type [bool] as type [string]", err)
```

Comment:
Unnecessary change in context of this PR

Reply:
Well....it's technically wrong, and I had to fix it in the test I created using this code as a base.  If control enters this block because `err == nil`, the dereference of `err.Error()` causes the test to segfault.  I can remove these fixes if you wish, though I found them to be useful since I triggered the analogous error in the tests I added and found the message to be more helpful than the segfault... :)

Reply:
Yes, it might be wrong, but suggest to send it as separate PR in this case ;) One PR -- one [logical] change, is what I strive for.

Reply:
Completely understood - I tend to fight to balance between "one PR, one logical change" and "fix what's broken."  I will gladly create a second PR to address these (admittedly rare!) test failures.  Thanks!